### PR TITLE
add custom opening and closing tags

### PR DESCRIPTION
**Pull request by supcik**
As discussed, here is a PR to allow for custom opening and closing tags

### CODE REVIEW

**Comment by supcik** [comment]
This PR addresses the issue #112

**Comment by supcik** [comment]
OK... I see that I failed the pre-commit.

I still have some issues with the pre-commit on my machine:

```
      × python setup.py egg_info did not run successfully.
      │ exit code: 1
      ╰─> [43 lines of output]
          /Users/.../.cache/pre-commit/repok0ned34x/py_env-default/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
```

and

```
pkgconfig.pkgconfig.PackageNotFoundError: md4c not found
```

I will work on it!

**Comment by supcik** [comment]
Solved. I had to install _md4c_ with `brew install md4c` (I am on MacOS)

**Comment by supcik** [comment]
I fixed most of the issues. I am still missing the Spanish translation for the README. I will make one with DeepL

**Comment by supcik** [comment]
The pre-commit passes on my machine now. Can you please run the workflow again?

File: README.md [hunk 1]
```diff
@@ -38,6 +38,18 @@ plugins:
 > Make sure that you define `include-markdown` before other plugins that could
  conflict, like [`mkdocs-macros-plugin`][mkdocs-macros-plugin-link].
 
+### Custom opening and closing tags
+
+The default opening and closing tags are `{%` and `%}`. You can change this
+default with the argments `opening_tag` and `closing_tag`:
```
Comment (mondeja) [inline README.md:44]:
```suggestion
The default opening and closing tags are `{%` and `%}`. You can change this
default with configuration fields `opening_tag` and `closing_tag`:
```

File: mkdocs_include_markdown_plugin/event.py [hunk 2]
```diff
@@ -586,20 +595,54 @@ def found_include_markdown_tag(match):
...
+    closing_tag='%}',
+):
+
+    # Escape special characters of the tags with a backslash.
+    # For example : '{%' will be escaped as '\{\%'
+    def escape(text):
+        TO_BE_ESCAPED = '.^$*+-?{}[]\\|():<>=!/#%,;'
+        return ''.join([('\\' if i in TO_BE_ESCAPED else '')+i for i in text])
+    escaped_opening_tag = escape(opening_tag)
+    escaped_closing_tag = escape(closing_tag)
```
Comment (mondeja) [inline mkdocs_include_markdown_plugin/event.py:623]: I think that you're reinveting the wheel here, you can use [`re.escape`](https://docs.python.org/3/library/re.html#re.escape):

```suggestion
    escaped_opening_tag = re.escape(opening_tag)
    escaped_closing_tag = re.escape(closing_tag)
```

**Review by mondeja — CHANGES_REQUESTED** [review]
Thanks, it's great! Just a couple of suggestions...

**Comment by supcik** [comment]
Thank you for the feedback. I did not know "re.escape", thank you for the tip!

**Comment by supcik** [comment]
Oups... I accepted your suggestion directly in the browser, so the pre-commit hooks did not run.

**Review by mondeja — APPROVED** [review]
Great!

**Pull request merged**